src/web100srv.c

```diff
@@ -1527,647 +1382,422 @@ void child_process(int parent_pipe, SSL_CTX *ssl_context, int ctlsockfd) {
-      if ((multiple == 1) && (mclients >= max_clients)) continue;
-
-      testing = 1;
-
-      // update all the clients so they have some idea that progress
-      // is occurring
-      if ((multiple == 0) && (waiting > 1)) {
-        j = waiting - 1;
-        tmp_ptr = head_ptr->next;
-        if (tmp_ptr == NULL) {
-          // no more clients waiting after the current one
-          waiting = 1;
-        }
-        while (tmp_ptr != NULL) {
-          log_println(3,
-                      "%d clients waiting, updating client %d "
-                      "testing will begin within %d minutes",
-                      (waiting - 1), tmp_ptr->pid, (waiting - j));
-          child_signal = waiting - j;
-          write(child_pipe[1], &child_signal, sizeof(int));
-          tmp_ptr = tmp_ptr->next;
-          j--;
-        }
-        
-        // more then 5 clients waiting, killing zombies, sending srv_queue
-        // messages to the child processes and ask them to run the status
-        // check. If the status check fails, the children will exit and then
-        // their entries will be cleaned up via the SIGCHLD handler.
-        // Similarly, a frozen child will have its watchdog timer go off, which
-        // will kill the process, which will invoke the SIGCHLD handler.
-        if ((waiting > 5) && (zombie_check == 0)) {
-          zombie_check = -1;
-          log_println(4, "More than 5 clients in the queue, "
-                      "remove zombies");
-          tmp_ptr = (struct ndtchild *)head_ptr;
-          // walk through this chain until the 4th client is reached, this helps
-          // prevent cases where the current client finishes and the next one begins
-          // before we are done looking
-          for (i = 0; i < 4; i++) { 
-            tmp_ptr = tmp_ptr->next;
-          }
-          while (tmp_ptr != NULL) {
-            i = SRV_QUEUE_HEARTBEAT;
-            tmp_pipe = tmp_ptr->pipe;
-            // Advance the pointer before the write(), because a SIGPIPE or
-            // SIGCHLD caused by the write could cause the memory pointed to by
-            // tmp_ptr to be cleared out from underneath us.
-            tmp_ptr = tmp_ptr->next;
-            write(tmp_pipe, &i, sizeof(int));
-          }
-          zombie_check = 0;
-          if (zombie_check > 0) {
-            waiting -= zombie_check;
-            zombie_check = 0;
-          }
-        }
-        
-      }
-
-      // multi_client:
-      if ((multiple == 1) && (mclients < max_clients)) {
-        if (mwaiting == 0) continue;
-        mwaiting--;
-        log_println(3,
-                    "Multi-client mode operation, dispatch a new "
-                    "client pid=%d.",
-                    mchild->pid);
-      } else {
-        log_println(3, "Telling client %d testing will begin now",
-                    head_ptr->pid);
-      }
+/**
+ * Retrieve the error message from OpenSSL and print it out.
```

Comment:
We should mention that we also exit the process because that's pretty important.